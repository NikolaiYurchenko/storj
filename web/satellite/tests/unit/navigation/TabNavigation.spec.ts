// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { shallowMount } from '@vue/test-utils';
import TabNavigation from '@/components/navigation/TabNavigation.vue';
import * as sinon from 'sinon';

const testRoutes = [
    {
        path: '/test1',
        name: 'Test1'
    },
    {
        path: '/test2',
        name: 'Test2'
    },
    {
        path: '/test3',
        name: 'Test3'
    },
    {
        path: '/test4',
        name: 'Test4'
    },
];

describe('TabNavigation.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(TabNavigation);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with navigation prop', () => {
        const wrapper = shallowMount(TabNavigation, {
            propsData: {navigation: testRoutes},
            stubs: ['router-link']
        });

        let wrapperArray = wrapper.findAll('.tab-navigation-container__item');

        for (let i = 0; i < wrapperArray.length; i++) {
            expect(wrapperArray.at(i).attributes().to).toBe(testRoutes[i].path);
        }
    });
});
