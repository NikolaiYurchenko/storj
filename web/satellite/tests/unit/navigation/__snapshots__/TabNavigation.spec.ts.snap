// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNavigation.vue renders correctly 1`] = `<div class="tab-navigation-container"></div>`;
